ts/WoltLabSuite/Core/Ui/Password.ts

```diff
@@ -0,0 +1,57 @@
+import DomChangeListener from "../Dom/Change/Listener";
+import * as Language from "../Language";
+
+const _knownElements = new WeakSet();
+
+export function setup(): void {
+  initElements();
+  DomChangeListener.add("WoltLabSuite/Core/Ui/Password", () => initElements());
+}
+
+function initElements(): void {
+  document.querySelectorAll("input[type=password]").forEach((input: HTMLInputElement) => {
+    if (!_knownElements.has(input)) {
+      initElement(input);
+    }
+  });
+}
+
+function initElement(input: HTMLInputElement): void {
+  _knownElements.add(input);
+
+  const inputAddon = document.createElement("div");
+  inputAddon.classList.add("inputAddon");
+  input.insertAdjacentElement("beforebegin", inputAddon);
+  inputAddon.appendChild(input);
+
+  const button = document.createElement("span");
+  button.title = Language.get("wcf.global.form.password.button.show");
+  button.classList.add("button", "inputSuffix", "jsTooltip");
+  button.setAttribute("role", "button");
+  button.tabIndex = 0;
+  button.setAttribute("aria-hidden", "true");
+  inputAddon.appendChild(button);
+
+  const icon = document.createElement("span");
+  icon.classList.add("icon", "icon16", "fa-eye-slash");
+  button.appendChild(icon);
+
+  button.addEventListener("click", () => {
+    toggle(input, button, icon);
+  });
+  button.addEventListener("keydown", (event) => {
+    if (event.key === "Enter" || event.key === " ") {
+      event.preventDefault();
+      toggle(input, button, icon);
+    }
+  });
+}
+
+function toggle(input: HTMLInputElement, button: HTMLElement, icon: HTMLElement): void {
+  icon.classList.toggle("fa-eye");
+  icon.classList.toggle("fa-eye-slash");
```

Comment:
Avoid using `toggle` when more logic independent of changing classes happens. Otherwise you might get inconsistent states.

I suggest to rewrite this whole method into a very explicit `if` for readability (avoids the use of ternaries).

```
if (input.type === "password") {
  icon.classList.add("fa-eye");
  icon.classList.add("fa-eye-slash");
  button.dataset.tooltip = Language.get(…);
  input.type = "text";
}
```